soc/arm/microchip_mec/mec1501/device_power.h

```diff
@@ -9,6 +9,84 @@
 
 #ifndef _ASMLANGUAGE
 
+/* Enable SoC control over peripherals only when drivers do not support
+ * power management
+ */
+#ifndef CONFIG_DEVICE_POWER_MANAGEMENT
+
+/* Comment out to use JTAG without interruptions.
+ * Beware this blocks PLL going off, hence should be enabled
+ * for power consumption measurements
+ * Note: To attach JTAG for any debug need to be performed with breakpoint
+ * prior to deep sleep entry.
+ */
+#define DEEP_SLEEP_JTAG
+
+/*
+ * CPU will spin up to DEEP_SLEEP_WAIT_SPIN_CLK_REQ times
+ * waiting for PCR CLK_REQ bits to clear except for the
+ * CPU bit itself. This is not necessary as the sleep hardware
+ * will wait for all CLK_REQ to clear once WFI has executed.
+ * Once all CLK_REQ signals are clear the hardware will transition
+ * to the low power state.
+ */
+
+/* #define DEEP_SLEEP_WAIT_ON_CLK_REQ_ENABLE */
+
+/* #define DEEP_SLEEP_CLK_REQ_DUMP */
+
+#define DEEP_SLEEP_WAIT_SPIN_CLK_REQ		1000
```

Comment:
We need to eliminate this spin loop.  It should not be necessary.  Once WFI is triggered the HW will turn off the PLL once all the PCR CLK_REQ bits are 0 after any blocks finish any work once they receive SLP_EN.

Reply:
I think you added it as debug mechanism though to keep it, 
Let's remove it
I actual prefer to use the "scratchpad" to save and dump later CLK_REQ values for blocking entry

Reply:
Agreed.  Your VBAT debug method is better.